### ISSUE TITLE
Scattered blocks instead of veins in world populators

### ISSUE THREAD

**Issue by cm4500** [PluginBugs/Issues-ItemsAdder#3323]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

cm4500

### What happened?

When using world populators, the custom blocks do not generate as contiguous veins but instead a collection of blocks near each other. 

I am wanting to create veins in a smaller but comparable size to that of perhaps diorite, granite, and andesite. But despite with tinkering with the "vein_blocks" and "chunk_veins" in the configs, I usually get "veins" with little to no actual blocks connected to each other. A picture is included below. 

This problem has existed since I started using ItemsAdder with 1.20.1, and plugin version ItemsAdder_3.6.2c. Though below, you'll see the current versions I am using.

### Steps to reproduce the issue

Use the world populator template provided on the ItemsAdder wiki. Replace the placeholders there with an actual block included in your pack, then adjust the values in "vein_blocks" and "chunk_veins" to your liking. My exact values are included in their own yml file below as a reference. 

### Server version

This server is running Paper version git-Paper-"a80d31b" (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: a80d31b)
You are running the latest version
Previous version: git-Paper-196 (MC: 1.20.1)

### ItemsAdder Version

ItemsAdder version 3.6.3-beta-7

### ProtocolLib Version

5.2.0-SNAPSHOT-679

### LoneLibs Version

1.0.45

### Problematic items yml configuration file (optional)

```shell
info:
  namespace: blocks
  enabled: true
worlds_populators:
  fuchsite_6080:
    block: blocks:fuchsite
    worlds:
      - world
    replaceable_blocks:
      - STONE
      - DIRT
      - ANDESITE
      - GRANITE
      - COBBLESTONE
      - GRAVEL
      - DEEPSLATE
      - COBBLED_DEEPSLATE
      - CALCITE
      - TUFF
    chunk_chance: 50
    max_height: 80
    min_height: 60
    vein_blocks: 20
    chunk_veins: 1
```

### Screenshots/Videos (you can drag and drop files or paste links)

![2024-01-12_00 41 34](https://github.com/PluginBugs/Issues-ItemsAdder/assets/127169368/518c0f39-ebd6-437e-94d2-5d081e544407)

**Comment by LoneDev6** [comment]
Can you travel to the same place you made the screenshot and press F3+G to show the chunk borders?
Thanks

**Comment by cm4500** [comment]
Yes, attached are two screenshots of the same chunk as above. Now I have worldedited out the stone and gravel surrounding it so it is easier to see the entire vein from two different angles.

![2024-01-15_12 30 21](https://github.com/PluginBugs/Issues-ItemsAdder/assets/127169368/e9c5887a-00e6-45f3-94c5-0bf13f12981f)
![2024-01-15_12 30 29](https://github.com/PluginBugs/Issues-ItemsAdder/assets/127169368/d011e80d-a500-4cce-97eb-ed6e365aca11)

**Comment by LoneDev6** [comment]
Will be checked, thank you.
This is very strange as during testing this didn't happen (long time ago).

**Comment by cm4500** [comment]
What is sort of interesting is that the surface decorators do not appear to have this problem. I am primarily use surface decorators for generating custom flowers across the world. And they all generate in contiguous veins. With ores generating scattered, it makes me wonder if these generating styles were swapped by mistake? If anything, both have their benefits. 

Is it possible to actual keep both generation styles and have it be configurable for each populator? For example, the setting can be "connected_vein: true" which will have the veins all connected like you'd like expect from vanilla ores, whereas if it is false, it will scatter in the manner of the original post? Just an idea. 

Thanks so much for a great plugin.

**Comment by LoneDev6** [comment]
> it makes me wonder if these generating styles were swapped by mistake? If anything, both have their benefits.

Impossible as they use a completely different code.

> Is it possible to actual keep both generation styles and have it be configurable for each populator? For example, the setting can be "connected_vein: true" ...

Well, this might be related to this other suggestion: #2435 (please ignore the customer being rude, this guy is known to harass me for years).

I will probably have to keep compatibility with the current ores generator to avoid old packs to have invalid generation values, because a new generation code might increase the number of blocks causing an unbalancement of the gameplay.

So the idea is to add a new property like `vein_type` which is defaulted to `LEGACY` and will access values like `CONNECTED`, `SCATTERED` or something similar.

I will do some tests and provide actual examples and proofs of concept for these new ideas.

Note that the current ores generation was tested extensively during a period of a few months during the default pack rebalancement process. This process involved me and some other players testing the gameplay to see if ores were hard or easy to find, to rebalance the gameplay and the result was satisfying, so this issue report confuses me a little.

## Here is more details about that:

### Vanilla generation
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/24ecc52b-5690-465a-bc98-f2f24f564cfb)

### My custom generation
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/e5b5ef26-2b1d-4a68-9274-3655048de6db)
![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/b06df8a7-02f9-4ff8-83cf-5cbf840904ee)

As you can see, based on my tests, they look very similar to vanilla generated veins.
